src/coreclr/jit/emitxarch.cpp

```diff
@@ -677,6 +589,156 @@ bool emitter::AreUpper32BitsSignExtended(regNumber reg)
 }
 #endif // TARGET_64BIT
 
+//------------------------------------------------------------------------
+// emitIsInstrWritingToReg: checks if the given register is being written to
+//
+// Arguments:
+//    id - instruction of interest
+//    reg - register of interest
+//
+// Return Value:
+//    true if the instruction writes to the given register.
+//    false if it did not, or if we can't safely determine.
+//
+bool emitter::emitIsInstrWritingToReg(instrDesc* id, regNumber reg)
+{
+    instruction ins = id->idIns();
+
+    // These are special cases since they modify one or more register(s) implicitly.
+    switch (ins)
```

Comment:
Some other instructions that might need to be covered based on the architecture manual + what's defined in `instrsxarch.h`:
* `movsb/d/q` - modifies `rdi` and `rsi` (this copies `ds:rsi` to `es:rdi` and increments or decrements `rdi` and `rsi`)
* `stosb/d/q` - modifies `rdi` (this copies `rax` to `es:rdi` and increments or decrements `rdi` )
* `r_movsb/d/q` - modifies `rdi`, `rsi`, and `rcx` (rep prefix executes the instruction `rcx` times)
* `r_stosb/d/q` - modifies `rdi` and `rcx` (rep prefix executes the instruction `rcx` times)
* `xchg` - modifies both registers, so may need special handling

We don't expose things like `cmpxchg8b/16b` or other "string" functions like `lodsb/d/q`, `cmpsb/d/q`, etc.

This doesn't include special cases for SIMD registers.

Reply:
I'll update this list accordingly.
As for SIMD, for now I'll only accept integer registers.

Reply:
`xchg` should be handled by the format since it doesn't look like they are implicit.

Reply:
@tannergooding  - The instructions that you mentioned, aren't those registers are encoded in `instrDesc*`? If yes, can we just use `id->idReg1() == reg`  to compare. I can see that sometimes we might have to check for `idReg2()` or `idReg3()`, but it might vary case to case basis.